include/networkit/distance/STSP.hpp

```diff
@@ -69,17 +90,79 @@ class STSP : public Algorithm {
     }
 
     /**
-     * Returns the distance from the source node to the target node
+     * If the target is a single node: returns the distance from the source node to the target
+     * node.
      * @return The distance from source to the target node.
      */
-    virtual edgeweight getDistance() const = 0;
+    edgeweight getDistance() const {
+        assureFinished();
+        return distance;
+    }
+
+    /**
+     * In case of multiple target nodes: returns the distance from the source node to the target
+     * nodes.
+     * @return Distances from the source to the target nodes.
+     */
+    const std::vector<edgeweight> &getDistances() const {
+        assureFinished();
+        return distances;
+    }
+
+    /**
+     * Sets the source node.
+     *
+     * @param newSource The new source node.
+     */
+    void setSource(node newSource) {
+        assert(G->hasNode(newSource));
+        source = newSource;
+    }
+
+    /**
+     * Sets the target node.
+     *
+     * @param newTarget The new target node.
+     */
+    void setTarget(node newTarget) {
+        assert(G->hasNode(newTarget));
+        target = newTarget;
+        targets.clear();
+    }
+
+    /**
+     * Sets the target nodes.
+     *
+     * @param targetsFirst,targetsLast Range of target nodes.
+     */
+    template <class InputIt>
+    void setTargets(InputIt targetsFirst, InputIt targetsLast) {
+        assert(std::all_of(targetsFirst, targetsLast, [&](auto u) { return G->hasNode(u); }));
+        targets.assign(targetsFirst, targetsLast);
```

Comment:
Should `target` be set back to `none`? Or otherwise should we drop variable `target` and use `targets`-vector in all cases (with implied changes for BidirectionalBFS/Dijkstra)?

Reply:
I think that none of the possibilities is satisfactory. It's weird for a class dealing with a single target to have it stored in a vector, while `target` is never used by `MultiTargetBFS/Dijkstra` -- so there is no point in setting it to `none`. I think that this issue can be solved by restructuring these algorithms.

Reply:
Agreed.